src/server/metadata/server/testing/metadata_test.go

```diff
@@ -1,36 +1,90 @@
 package testing
 
 import (
-	"strings"
 	"testing"
 
+	"github.com/google/go-cmp/cmp"
 	"github.com/pachyderm/pachyderm/v2/src/internal/dockertestenv"
 	"github.com/pachyderm/pachyderm/v2/src/internal/pachd"
 	"github.com/pachyderm/pachyderm/v2/src/internal/pctx"
 	"github.com/pachyderm/pachyderm/v2/src/internal/testpachd/realenv"
 	"github.com/pachyderm/pachyderm/v2/src/metadata"
+	"github.com/pachyderm/pachyderm/v2/src/pfs"
+	"google.golang.org/protobuf/testing/protocmp"
 )
 
-func TestTestPachd(t *testing.T) {
+func TestRealEnv(t *testing.T) {
```

Comment:
It might be worth promoting this logic up to `src/server/metadata` (for logic) and `src/server/metadata/server` (for gRPC handling).

Reply:
So the integration tests always live in server/testing.  This one actually uses PFS to create a project, so while the tests could go in //src/metadata, it's more "normal" to have them here instead.  This convention originates from all the API servers that have circular dependencies, though, so isn't actually necessary for anything other than PFS or PPS.  But consistency...